appserver/node-express/lib/db-client/qnaDoc/patch.js

```diff
@@ -66,6 +116,24 @@ module.exports = function (txid, spec) {
     );
   };
 
+  /**
+   * Handle the post of a new answer to a Samplestack question. The posted
+   * answer is merged with an answer template object, a unique ID, contributor
+   * information, and the current time. The resulting object is written to the
+   * MarkLogic database. During the write, the answer count for the question is
+   * incremented and the last activity date is updated.
+   *
+   * @param  {String} txid The transaction ID.
+   * @param  {Object} contributor Contributor object. Example:
+   *   {"id":"cf99542d-f024-4478-a6dc-7e723a51b040",
+   *    "displayName":"JoeUser"}
+   * @param  {String} questionId The question ID.
+   * @param  {Object} spec The answer content from the request. Example:
+   *   {"comments":[],
```

Comment:
Someone seems to have deleted a discussion about comments... the concern I have is that if an empty comments array needs to be include in the document when it is loaded, it is very much this function's concern to provide that, and not the router, as it's a storage-level problem, not an app-logic problem. Let's check whether or not the empty array is required, and if so, this function should assure that it is present before attempting to store the document.